feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c

```diff
@@ -287,27 +288,48 @@ bool target_stats_survey_convert(radio_entry_t *radio_cfg, radio_scan_type_t sca
 		memset(data_old, 0, sizeof(*data_old));
 	} else if ((data_new->duration_ms <= data_old->duration_ms) &&
 			(data_new->chan_busy <= data_old->chan_busy)) {
-		/*
-		 * Take care of survey data from CHAN_INFO events which are Read-On-Clear
-		 * in nature, whereas survey data from BSS_CHAN_INFO events are Read only.
-		 * Chan_info event down in the driver updates only the duration, noise floor
-		 * and the chan_busy data.
-		 */
-		data_new->duration_ms += data_old->duration_ms;
-		data_new->chan_busy += data_old->chan_busy;
+		LOGD("target_stats_survey_convert: for_report:%d, scan_type:%d, chan:%d,"
+			"oldSurveyP:%p, newChanDur:%llu, oldDur:%llu, newBusy:%llu, oldBusy:%llu,"
+			" new_nf:%u, old_nf:%u, scInt:%d\n",
+			for_report, scan_type, data_new->info.chan, data_old, data_new->duration_ms,
+			data_old->duration_ms, data_new->chan_busy, data_old->chan_busy,
+			data_new->chan_noise, data_old->chan_noise, scan_interval);
+
+		if ((data_new->duration_ms <= 150) &&
+			(data_new->duration_ms <= (uint64_t)scan_interval)) {
```

Comment:
As discussed, the above condition would not work as expected. We have to come up with a different solution.